src/coreclr/jit/morph.cpp

```diff
@@ -5465,11 +5465,21 @@ GenTree* Compiler::fgMorphArrayIndex(GenTree* tree)
 
     GenTree* cns = gtNewIconNode(elemOffs, TYP_I_IMPL);
 
+// We're going to create a platform-specific address:
+//
+// XArch:   "arrRef + (index + elemOffset)"
+// ArmArch: "(arrRef + elemOffset) + index" // both ADD are BYREF here
+//
+// It'll allow us to hoist/CSE the invariant part which is "arrRef + elemOffset" on ArmArch.
+// We don't really need it on XArch because of more powerful addressing modes such as [base + index*scale + offset]
+//
```

Comment:
You can use the ` CLANG_FORMAT_COMMENT_ANCHOR;` hack here to fix the comment indenting.